tests/cross/src/test/scala/tests/hover/HoverTermSuite.scala

```diff
@@ -305,7 +333,29 @@ class HoverTermSuite extends BaseHoverSuite {
     """|final def withFilter(p: Int => Boolean): Option[Int]#WithFilter
        |""".stripMargin.hover,
     compat = Map(
-      "3" -> "val <local a$>: (x: Int): Boolean".hover
+      "3" ->
+        """|Boolean
+           |def withFilter(p: A => Boolean): Option.this.WithFilter
+           """.stripMargin.hover
+    )
+  )
+
+  check(
+    "for-method",
+    """
+      |object a {
+      |  for {
+      |    x <- <<List(1).headOp@@tion>>
+      |  } yield x
+      |}
+      |""".stripMargin,
+    """|override def headOption: Option[Int]
+       |""".stripMargin.hover,
+    compat = Map(
+      "3" ->
+        """|Option[Int]
```

Comment:
```suggestion
```

Reply:
Shouldn't we show expression type here?
I'm actually not sure in which case should we show the expression type (when it's not range hover) I thought the case we should show the expression type is like https://github.com/dos65/metals/blob/fab0ac80abbf5af40ecf566df13571692310713e/tests/cross/src/test/scala/tests/hover/HoverDocSuite.scala#L8-L60

- The symbol is a method, whose return type in the doc is written with type parameter like `java.util.List[T]`
- On the other hand, expression type has inferred concrete type in the type parameter like `java.util.List[Int]`

Reply:
You're probably right in this case, though it would be awesome if we show the exact types in hover like in Scala 2, but it's not what is done currently. Needs to be fixed separately